File: CycleChesscpp/app/src/main/cpp/pieces/utils/color.cpp
#include "color.h"

Color getAnotherColor(const Color color) {
    switch (color)
    {
    case Color::WHITE:
        return Color::BLACK;
    case Color::BLACK:
        return Color::WHITE;
    default:
        // error
        return Color::NO_COLOR;
    }
}

Colors getAllColors() {
    return {
        Color::WHITE,
        Color::BLACK
    };
}

std::string getTurnColorFen(const Color color) {
    // TODO переделать свичи так, чтобы они покрывали все
    // случаи, тогда компилятору будет лучше
    switch (color)
    {
    case Color::WHITE:
        return "w";
    case Color::BLACK:
        return "b";
    default:
        // TODO error need log
        return " ";
    }
}

Color getTurnColorFromFen(std::string fen) {
    if (fen == "w") {
        return Color::WHITE;
    }
    if (fen == "b") {
        return Color::BLACK;
    }
    // TODO error need log
    return Color::NO_COLOR;
}

File: CycleChessBot/bot/chess_game/color.py
from enum import Enum

class Color(Enum):
    NO_COLOR = 0
    WHITE = 1
    BLACK = 2


    def __str__(self) -> str:
        return str(self.value)

def get_another_color(color: Color) -> Color:
    return Color.BLACK if color == Color.WHITE else Color.WHITE

def get_all_colors() -> list[Color]:
    return [Color.WHITE, Color.BLACK]
